torch/csrc/utils/tensor_new.cpp

```diff
@@ -146,4 +159,25 @@ Tensor tensor_new(const Type& type, PyObject* args, PyObject* kwargs) {
   throw std::runtime_error("new(): invalid arguments");
 }
 
+static const Tensor & set_requires_grad(const Tensor &self, bool requires_grad) {
+  torch::autograd::Variable(self).get()->_requires_grad = requires_grad;
+  return self;
+}
+
+Tensor variable_data_factory(const Type& type, PyObject* args, PyObject* kwargs) {
+  static PythonArgParser parser({
+    "new(Tensor other, *, bool requires_grad=False)",
+    "new(PyObject* data, *, int64_t device=-1, bool requires_grad=False)",
+  });
+
+  PyObject* parsed_args[3];
+  auto r = parser.parse(args, kwargs, parsed_args);
+  if (r.idx == 0) {
+    return set_requires_grad(new_with_tensor(type, r.tensor(0)), r.toBool(1));
```

Comment:
This currently will create an "alias" of the tensor "other". I think it would be better to copy the data to match NumPy semantics and the `new_from_data` overload.

Reply:
agreed.